### PR TITLE
feat(aitools): record install error categories from CLI JSON output

### PR DESCRIPTION
**Pull request by rclarey**
## Changes

Run `aitools install` with `--output json` and parse the result so the AITOOLS_INSTALL telemetry event carries the CLI's error categories: a top-level `globalErrorCategory` and a per-agent `agentErrors` map, both category tokens only (never the free-form error/message strings)

## Tests

updated / added unit tests

### CODE REVIEW

**Comment by rugpanov** [comment]
🤖 Integration tests ❌ **failed** for `839fc721`.
[View run](https://go/deco-tests/34473556587)

**Comment by rugpanov** [comment]
🤖 Integration tests ✅ **passed** for `e2ed1a14`.
[View run](https://go/deco-tests/34850986177)

File: packages/databricks-vscode/src/aitools/AiToolsManager.ts [hunk 1]
```diff
@@ -501,6 +492,68 @@ export class AiToolsManager implements Disposable {
...
+            const errorCode =
+                "code" in error && typeof error.code === "number"
+                    ? error.code
+                    : null;
+            const message = describeAiToolsInstallFailure(output);
+            if (message) {
+                // The CLI reported the install failed in its JSON (non-zero exit, but
+                // a parseable result). Under `--output json` the CLI's stderr is empty,
+                // so log the folded detail here, then surface it to the UI by throwing.
+                logging.NamedLogger.getOrCreate(Loggers.Extension).error(
```
Comment (misha-db) [inline packages/databricks-vscode/src/aitools/AiToolsManager.ts:546]: Duplicate logging per failure?
CliWrapper.aitoolsInstall logs the raw error and `runInstallCli` logs the folded JSON, producing two log entries per failure.

File: packages/databricks-vscode/src/telemetry/constants.ts [hunk 2]
```diff
@@ -424,6 +426,25 @@ export class EventTypes {
...
+        agentErrors: {
+            comment:
+                "Per-agent install errors as a map of agent id to its categorical error, for " +
+                'each requested agent that did not install (failed or skipped) — e.g. {"codex":' +
+                '"UNSUPPORTED_SCOPE"}. Keyed on the agent id so a category can be attributed to a ' +
+                "specific agent (exploded into one row per agent downstream). Categories are the " +
+                "CLI's closed set; an unrecognised token collapses to 'other'. Never the free-form " +
+                "per-agent message. JSON-stringified into a property by recordEvent. Omitted when " +
+                "every requested agent installed, or when the bundled CLI predates JSON install output",
+        },
```
Comment (misha-db) [inline packages/databricks-vscode/src/telemetry/constants.ts:447]: comments say an unrecognised token "collapses to 'other'", but summarizeAiToolsInstallErrors emits the CLI's raw categorym no client-side normalization. So a new CLI category reaches telemetry as is.

File: packages/databricks-vscode/src/cli/CliWrapper.ts [hunk 3]
```diff
@@ -223,6 +223,94 @@ export interface AiToolsListResult {
...
+    };
+    const hasAgents = Array.isArray(rec.agents);
+    const hasError =
+        typeof rec.error === "string" || typeof rec.error_category === "string";
+    if (!hasAgents && !hasError) {
+        return undefined;
+    }
+    return {
+        ...(obj as AiToolsInstallOutput),
+        agents: hasAgents ? (rec.agents as AiToolsInstallAgentResult[]) : [],
```
Comment (misha-db) [inline packages/databricks-vscode/src/cli/CliWrapper.ts:310]: nit: checks field presence only, then and casts to AiToolsInstallOutput. A doc like {"error_category":"X","error":123} passes and returns error as a number typed as string

Should we be more defensive?

File: packages/databricks-vscode/src/aitools/aiToolsInstallOutput.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    if (output === undefined) {
+        return undefined;
+    }
+    const lines: string[] = [];
+    if (output.error) {
+        lines.push(output.error);
+    }
+    for (const agent of output.agents) {
+        if (agent.status !== "installed") {
+            const detail = agent.message ?? "failed to install";
```
Comment (misha-db) [inline packages/databricks-vscode/src/aitools/aiToolsInstallOutput.ts:28]: Does this treat every non installed agent, including a legitimate skipped ones as a failure (with message "failed to install")?

**Review by misha-db — CHANGES_REQUESTED** [review]

File: packages/databricks-vscode/src/aitools/aiToolsInstallOutput.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        return {};
+    }
+    const summary: AiToolsInstallErrorSummary = {};
+    if (output.error_category !== undefined) {
+        summary.globalErrorCategory = output.error_category;
+    }
+    const agentErrors: Record<string, string> = {};
+    for (const agent of output.agents) {
+        if (agent.error_category !== undefined) {
+            agentErrors[agent.name] = agent.error_category;
```
Comment (rugpanov) [inline packages/databricks-vscode/src/aitools/aiToolsInstallOutput.ts:54]: **Normalize `error_category` to a closed set before it reaches telemetry**

This copies the CLI's `error_category` verbatim into both `globalErrorCategory` and every `agentErrors` value. But the event's own doc comment in `constants.ts` promises *"a closed category token … an unrecognised token collapses to `'other'`"*, and nothing here (or downstream, as far as I can see) actually performs that collapse.

This is inconsistent with how we handle CLI-sourced strings elsewhere in telemetry:
- `telemetry/README.md` (Privacy): *"Only categorical/enum values and durations — no file paths, cluster names or IDs … or user content."*
- `envKey` is the direct precedent — it pattern-checks the CLI value to *"keep a closed vocabulary; anything unrecognised collapses to `'other'`"*, precisely because minimally-parsed CLI JSON can carry user-chosen content.
- Other categorical props are typed as closed unions (e.g. `type?: "dabs" | "legacy" | "unknown"`); these two are `string` / `Record<string, string>`, i.e. open.

Two concrete risks:
1. **High cardinality.** An unbounded category value — especially `agentErrors`, which is a whole map JSON-stringified into a single property value — puts unbounded distinct values on the wire, which is exactly what strains/breaks telemetry ingestion and cost (separate from privacy).
2. **Leak.** If the CLI ever emits something free-form in `error_category` (a bug, or an older/newer CLI), it flows straight through unchecked.

Suggestion: define the known category set client-side and collapse anything else to `"other"` (mirroring `envKey`/`UNRECOGNISED_ENV_KEY`), and type the fields as that union rather than open `string`. If the collapse is actually meant to happen server-side, could we update the comment to say so and point at where it happens?

File: packages/databricks-vscode/src/aitools/AiToolsManager.ts [hunk 6]
```diff
@@ -501,6 +492,68 @@ export class AiToolsManager implements Disposable {
...
+        const {output, error} = await this.cli.aitoolsInstall(
+            scope,
+            this.cwdForScope(scope),
+            token,
+            agents
+        );
+
+        const summarizedErrors = summarizeAiToolsInstallErrors(output);
+        recordEvent({
+            result: error !== undefined ? "error" : "success",
```
Comment (rugpanov) [inline packages/databricks-vscode/src/aitools/AiToolsManager.ts:521]: **`result` and the recorded categories can disagree — a zero-exit "success" can carry errors**

`result` is derived only from the process exit (`error !== undefined ? "error" : "success"`), while `summarizeAiToolsInstallErrors(output)` attaches categories independently — and it keys on `error_category` for *any* agent regardless of `status`.

So a request the CLI **skips** on a clean exit — e.g. `codex` in `project` scope (`status: "skipped", error_category: "UNSUPPORTED_SCOPE"`, the `perAgentSkip` shape), which exits 0 — records `result: "success"` **with** `agentErrors: { codex: "UNSUPPORTED_SCOPE" }`. A dashboard counting events with a non-empty `agentErrors` as failures would overcount, and the success flag would contradict the category data.

This may be intentional (the `agentErrors` comment does say "failed or skipped") — if so, could we make it explicit and add a test? Today every category test pairs categories with a non-zero `error`, so the zero-exit-with-skip path is uncovered. If skips shouldn't count, gating on `status === "failed"` (and/or on `error !== undefined`) keeps the two in sync. Minor related nit: `summarizeAiToolsInstallErrors` records a category even for `status: "installed"` agents, since it never checks `status`.

**Review by anton-107 — APPROVED** [review]
Reviewed the install JSON parsing, failure handling, and telemetry path. No blocking issues found.

File: packages/databricks-vscode/src/aitools/AiToolsManager.ts (hunk 1 above)
Comment (rclarey) [reply packages/databricks-vscode/src/aitools/AiToolsManager.ts]: Yeah as the comment mentions we log here because `stderr` is empty with `--output json`, so we need special handling for that case. I didn't want to miss non-JSON reported errors so I kept the existing log in `CliWrapper.aitoolsInstall`, but you're right maybe having two lines is confusing. I'll handle both in AiToolsManager 👍

File: packages/databricks-vscode/src/telemetry/constants.ts (hunk 2 above)
Comment (rclarey) [reply packages/databricks-vscode/src/telemetry/constants.ts:436]: Missed this when I removed category parsing, will update

File: packages/databricks-vscode/src/aitools/aiToolsInstallOutput.ts (hunk 4 above)
Comment (rclarey) [reply packages/databricks-vscode/src/aitools/aiToolsInstallOutput.ts:28]: No, `output.agents` only contains the agents that were attempted (i.e. the ones the user selected in the picker), so that's why if the status is not `installed` then it means it failed. Also this function is only called for non-zero exit codes, so there is no legitimate skips here.

File: packages/databricks-vscode/src/aitools/aiToolsInstallOutput.ts (hunk 5 above)
Comment (rclarey) [reply packages/databricks-vscode/src/aitools/aiToolsInstallOutput.ts:54]: This is intentional. The error categories are already an enum emitted from the CLI so this extension just treats them as opaque and forwards to telemetry to we can associate the error category with other extension telemetry. Parsing and emitting the same enums the CLI outputs just adds error surface where the two can drift apart on a CLI bump

File: packages/databricks-vscode/src/aitools/AiToolsManager.ts (hunk 6 above)
Comment (rclarey) [reply packages/databricks-vscode/src/aitools/AiToolsManager.ts:524]: Yes this is intentional. The CLI is the source of truth for error categorization so this PR intentionally doesn't add it's own logic and forwards any error categories it receives. I will add a test for 0 exit + skip error category to codify that this is intentional.

> e.g. codex in project scope (status: "skipped", error_category: "UNSUPPORTED_SCOPE", the perAgentSkip shape), which exits 0

This is only true when codex is auto-detected. When `--agents` is passed (which this extension always does) then `UNSUPPORTED_SCOPE` because a hard error and the CLI exits 1. 

> A dashboard counting events with a non-empty agentErrors as failures would overcount

This is a dashboard error, exit code is the only indicator of "success" vs "error". The reason we always report error categories from the CLI, even for 0 exit with skips, is because otherwise we wouldn't get any telemetry about those skips (e.g. "How often is codex auto-detected but skipped for a project scoped installation?")

**Comment by rugpanov** [comment]
🤖 Integration tests running for `0e776545` — ⏳.
[View run](https://github.com/databricks-eng/eng-dev-ecosystem/actions/workflows/vscode-isolated-pr.yml)

**Review by misha-db — APPROVED** [review]
